### ISSUE TITLE
setStateDelayed only fires the first and last call for the same state-id

### ISSUE THREAD

**Issue by systemofapwne** [ioBroker/ioBroker.javascript#400]
I recently wrote code similar to
```
setStateDelayed("TheState",val1,1000);
setStateDelayed("TheState",val2,2000);
setStateDelayed("TheState",val3,3000);
setStateDelayed("TheState",val4,4000);
```
Whatever I tried, but only the very first setStateDelayed got "val1" applied and the last one with "val4" (both of type string). The setStateDelayed inbetween have been completely ignored.

For reasons: The code I used
```
var g_Colors = {}; //Stores previous colors
var g_Dimm = {}; //Stores previous dimm values
var g_Power = {}; //Stores previous power state
var g_dt = 2000; //Delay between each color change
function Notification(color){
    //Read in present colors, enable fading and fade to Black
    g_LEDs.forEach((v) =>{
        g_Colors[v] = getState(v+".Color").val;
        g_Dimm[v] = getState(v+".Dimmer").val;
        g_Power[v] = getState(v+".POWER").val;
        setState(v+".Fade",true);
        setState(v+".Speed",3);
        setStateDelayed(v+".Color","#00FF00",0); //Gets applied
        setStateDelayed(v+".Color","#FF0000",2000); //Nope
        setStateDelayed(v+".Color","#0000FF",4000); //Nope
        setStateDelayed(v+".Color","#000000",6000);  //Nope
        setStateDelayed(v+".Color","#FF0000",8000); //Gets applied
       //... additional code ...
```

**Comment by BuZZy1337** [comment]
https://github.com/ioBroker/ioBroker.javascript/blob/master/docs/en/javascript.md#setstatedelayed

Use "false" at the end of the parameters.. That means "do not override the existing setStateDelayed for the given state..

```js
setStateDelayed("TheState",val1,1000);
setStateDelayed("TheState",val2,2000,false);
setStateDelayed("TheState",val3,3000,false);
setStateDelayed("TheState",val4,4000,false);
```

**Comment by systemofapwne** [comment]
I was under the impression, that clearRunning was false by default. Yet, but even if is NOT false by default does not explain, why the **first** setStateDelayed works, nothing inbetween and then the **last** one again. Sure, in my "real" code, the first call has a delay of 0 (this was just for testing) and one might argue, that it gets called instantly (which I anway doubt). But it also failed, when it was like 100 ms or a second: Only the first and last call and non inbetween worked.

**Comment by AlCalzone** [comment]
0ms might mean that the callback is invoked before the second invocation clears the timer again. Try with a small number larger than 0ms, e.g. 20ms to be sure

**Comment by systemofapwne** [comment]
As I said, it happened with a delay greater than 0 ms: It initisally was 100 ms.

**Comment by AlCalzone** [comment]
Sorry I misread

**Comment by systemofapwne** [comment]
It must have been a coincidence, why it also failed for me with a delay on the first call, when I tried it yesterday. Setting clearRunning to false index fixes the problem.

**Comment by BuZZy1337** [comment]
As AlCalzone wrote:
0ms fires the setStateDelayed immediately and runs out before the next setStateDelayed is invoked or even "registered".. Thats more or less the same as you would just use setState();

Example:
```js
setStateDelayed("javascript.0.TestDP", "#00FF00", 0, function () {
    log("Delay 1 fired!");
}); //Gets applied
setStateDelayed("javascript.0.TestDP", "#00FF00", 2000, function () {
    log("Delay 2 fired!");
}); //Nope
setStateDelayed("javascript.0.TestDP", "#00FF00", 4000, function () {
    log("Delay 3 fired!");
}); //Nope
setStateDelayed("javascript.0.TestDP", "#00FF00", 6000, function () {
    log("Delay 4 fired!");
});  //Nope
setStateDelayed("javascript.0.TestDP", "#00FF00", 8000, function () {
    log("Delay 5 fired!");
}); //Gets applied
```
Output:
```
11:39:54.102 | info | javascript.0 script.js.Testskripte.Skript1: Delay 1 fired!
11:40:02.101 | info | javascript.0 script.js.Testskripte.Skript1: Delay 5 fired!
```

**Comment by systemofapwne** [comment]
Yes, exactly. I initially had the first delay on 100 ms/1000ms. I also discovered problems there and I must then have changed the delay to 0 for testing purposes and then later made the wrong assumption, that the first and last setStateDelayed only have been called. The solution is setting clearRunning to false. There was no bug, just my wrong assumption of "clearRunning = default:false" and improper conclusion. Therefore I closed it again.
